src/cmd/singularity/cli/actions.go

```diff
@@ -255,22 +255,34 @@ func execWrapper(cobraCmd *cobra.Command, image string, args []string, name stri
 		}
 	}
 
-	if !IsCleanEnv {
-		for _, env := range os.Environ() {
-			e := strings.SplitN(env, "=", 2)
-			if len(e) != 2 {
-				sylog.Verbosef("can't process environment variable %s", env)
-				continue
-			}
-			if e[0] == "HOME" {
-				if !NoHome {
-					generator.AddProcessEnv(e[0], engineConfig.GetHome())
-				} else {
-					generator.AddProcessEnv(e[0], "/")
-				}
+	// Copy and cache environment
+	environment := os.Environ()
+
+	// Clean environment
+	for _, env := range environment {
+		e := strings.SplitN(env, "=", 2)
+		if len(e) != 2 {
+			sylog.Verbosef("can't process environment variable %s", env)
+			continue
+		}
+
+		// Transpose environment
+		if strings.HasPrefix(env, "SINGULARITYENV_") {
+			te := strings.Split(strings.TrimPrefix(env, "SINGULARITYENV_"), "=")
```

Comment:
Here we should handle checking the len of the split string. Also, we should just do:

```
e[0] = te[0]
```

and not write `continue`. This way the line 285 code still handles this. We're just simply changing the key